src/components/OrderList.ts

```diff
@@ -0,0 +1,172 @@
+import Coffee from '../model/Coffee';
+import Option from '../model/Option';
+import CoffeeService from '../services/CoffeeService';
+import { findById } from '../utils';
+
+type Order = {
+  id: number;
+  editable: boolean;
+  coffee: Coffee;
+  options: Option[];
+  orderedAt: Date;
+  deliveredAt: Date | null;
+};
+
+export type OrdersState = { orders: Order[] };
+
+class OrderList implements Component, Observable {
+  private $root: HTMLElement;
+  private state: OrdersState = { orders: [] };
+  private observers$ = new Set();
+
+  private static uniqueIdCnt = 0;
+
+  constructor($root: HTMLElement | null) {
+    if (!$root) throw new Error('root element is required to render');
+    this.$root = $root;
+    this.init();
+  }
+
+  init() {
+    this.setEvent();
+  }
+
+  subscribe<OrdersState>(_cb: (s: OrdersState) => void) {
+    const callback = _cb;
+    this.observers$.add(callback);
+    return {
+      unsubscribe: () => this.observers$.delete(callback),
+    };
+  }
```

Comment:
```suggestion
  subscribe<OrdersState>(callback: (s: OrdersState) => void) {
    this.observers$.add(callback);
    return { unsubscribe: () => this.observers$.delete(callback) };
  }
```
이렇게 선언할수 있을것 같은데 `const callback = _cb;`가 있는 다른 이유가 있을까요~?

Reply:
큰 이유는 없었고 제가 왜 이렇게 했는지 모르겠네요 ㅠㅠ 바로 수정했어요!